src/vault/client/http.clj

```diff
@@ -130,6 +133,21 @@
   (reset! (:auth client) {:client-token (str/trim token)}))
 
 
+(defn- authenticate-wrap-token!
+  "Updates the token ref by making an unwrap request that returns the auth token."
+  [client credentials]
+  (api-auth!
+    "wrapped token"
+    (:auth client)
+    (do-api-request :post (str (:api-url client) "/v1/sys/wrapping/unwrap")
```

Comment:
Ah, you could still refactor this into a common `unwrap-secret` function, but you are correct about leaving the main `api-request` function unchanged.